sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/SessionReceiverManager.cs

```diff
@@ -348,15 +355,15 @@ private async Task RenewSessionLock()
                             TaskContinuationOptions.ExecuteSynchronously,
                             TaskScheduler.Default)
                         .ConfigureAwait(false);
-                    if (Receiver.IsClosed || delayTask.IsCanceled)
```

Comment:
This should not be necessary as we await the RenewSessionLock task before calling Dispose on the receiver.